Mathlib/LinearAlgebra/FiniteDimensional.lean

```diff
@@ -922,6 +922,13 @@ theorem injective_iff_surjective [FiniteDimensional K V] {f : V →ₗ[K] V} :
         this).injective⟩
 #align linear_map.injective_iff_surjective LinearMap.injective_iff_surjective
 
+lemma injOn_iff_surjOn {p : Submodule K V} [FiniteDimensional K p]
+    {f : V →ₗ[K] V} (h : ∀ x ∈ p, f x ∈ p) :
```

Comment:
This is defeq, but possibly easier to work with at the caller:
```suggestion
    {f : V →ₗ[K] V} (h : p \le p.comap f) :
```

Reply:
I was tempted to use this spelling but I opted to follow that of [LinearMap.restrict](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Basic.html#LinearMap.restrict). I don't think it matters where I currently use it.